lucet-wiggle/generate/src/config.rs

```diff
@@ -3,37 +3,79 @@ use syn::{
     braced,
     parse::{Parse, ParseStream},
     punctuated::Punctuated,
-    Error, Ident, Result, Token,
+    Error, Result, Token,
 };
 use wiggle_generate::config as w;
 
+mod kw {
+    syn::custom_keyword!(witx);
+    syn::custom_keyword!(witx_literal);
+    syn::custom_keyword!(ctx);
+    syn::custom_keyword!(errors);
+    syn::custom_keyword!(constructor);
+    syn::custom_keyword!(pre_hook);
+    syn::custom_keyword!(post_hook);
+}
+
 #[derive(Debug, Clone)]
 pub struct Config {
     pub wiggle: w::Config,
     pub constructor: TokenStream,
+    pub pre_hook: Option<TokenStream>,
+    pub post_hook: Option<TokenStream>,
 }
 
 #[derive(Debug, Clone)]
 pub enum ConfigField {
     Wiggle(w::ConfigField),
     Constructor(TokenStream),
+    PreHook(TokenStream),
+    PostHook(TokenStream),
 }
 
 impl Parse for ConfigField {
     fn parse(input: ParseStream) -> Result<Self> {
-        let id: Ident = input.parse()?;
-        let _colon: Token![:] = input.parse()?;
-        match id.to_string().as_ref() {
-            "constructor" => {
-                let contents;
-                let _lbrace = braced!(contents in input);
-                Ok(ConfigField::Constructor(contents.parse()?))
-            }
-            _ => Ok(ConfigField::Wiggle(w::ConfigField::parse_pair(
-                id.to_string().as_ref(),
-                input,
-                id.span(),
-            )?)),
+        let lookahead = input.lookahead1();
+        if lookahead.peek(kw::constructor) {
+            input.parse::<kw::constructor>()?;
+            input.parse::<Token![:]>()?;
+            let contents;
+            let _lbrace = braced!(contents in input);
```

Comment:
I'm curious about the assignment to `_lbrace` here.  Is the assignment necessary, or is the variable name used simply to convey meaning?

Reply:
`let _ = braced!(...)` would be sufficient, but the token that macro returns is specifically the left brace token, which could be used for span information. This is basically the same as the example in the `braced!` docs except I'm not preserving any of the tokens I don't actually need: https://oschwald.github.io/maxminddb-rust/syn/macro.braced.html